profiles/schema.py

```diff
@@ -1408,6 +1409,54 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteMyProfileMutation()
 
 
+class DeleteServiceDataMutation(relay.ClientIDMutation):
+    class Input:
+        authorization_code = graphene.String(
+            required=True,
+            description=(
+                "OAuth/OIDC authorization code. When obtaining the code, it is required to use "
+                "service and operation specific GDPR API scopes."
+            ),
+        )
+        service_name = graphene.String(
+            required=True,
+            description=("The name of the service the data should be removed from"),
+        )
+        dry_run = graphene.Boolean(
+            required=False,
+            description="Can be used to see if the date can be removed from the service. Default is False.",
+        )
+
+    @classmethod
+    @login_and_service_required
+    def mutate_and_get_payload(cls, root, info, **input):
+        try:
+            profile = Profile.objects.get(user=info.context.user)
+        except Profile.DoesNotExist:
+            raise ProfileDoesNotExistError("Profile does not exist")
+
+        if not info.context.service.has_connection_to_profile(profile):
+            raise PermissionDenied(
+                _("You do not have permission to perform this action.")
+            )
+
+        service_connections = profile.effective_service_connections_qs().filter(
+            service__name=input["service_name"]
+        )
+
+        if not service_connections:
+            raise ConnectedServiceDeletionFailedError("Service connection not found")
```

Comment:
Maybe there should be a dedicated error for this case, where no service connection exists for the named service. Now this same error is used for two different scenarios:

1. Deletion was tried but it failed for some reason (this is the old case)
1. No service connection for the given service name was found (this new case)

My reasoning is that it's possible that these cases need to be distinguished in the UI because the next step the user needs to take is different. In case 1 the user may want to try again or if it keeps failing, contact user support or something like that. In case 2 the user doesn't need to take next steps. The named service doesn't have any data about the user (as far as we know anyway) and that's the end result the user wanted.

Reply:
Good idea. Added a new exception.